packages/syft/src/syft/service/worker/utils.py

```diff
@@ -248,6 +254,151 @@ def run_workers_in_threads(
     return results
 
 
+def create_kubernetes_pool(
+    runner: KubernetesRunner,
+    worker_image: SyftWorker,
+    pool_name: str,
+    replicas: int,
+    queue_port: int,
+    debug: bool,
+):
+    pool = None
+    error = False
+
+    try:
+        print(
+            "Creating new pool "
+            f"name={pool_name} "
+            f"tag={worker_image.image_identifier.full_name_with_tag} "
+            f"replicas={replicas}"
+        )
+        pool = runner.create_pool(
+            pool_name=pool_name,
+            tag=worker_image.image_identifier.full_name_with_tag,
+            replicas=replicas,
+            env_vars={
+                "SYFT_WORKER": "True",
+                "DEV_MODE": f"{debug}",
+                "QUEUE_PORT": f"{queue_port}",
+                "CONSUMER_SERVICE_NAME": pool_name,
+                "N_CONSUMERS": "1",
+                "CREATE_PRODUCER": "False",
+                "INMEMORY_WORKERS": "False",
+            },
+        )
+    except Exception as e:
+        error = True
+        return SyftError(message=f"Failed to start workers {e}")
+    finally:
+        if error and pool:
+            pool.delete()
+
+    return runner.get_pods(pool_name=pool_name)
+
+
+def scale_kubernetes_pool(runner: KubernetesRunner, pool_name: str, replicas: int):
+    pool = runner.get_pool(pool_name)
+    if not pool:
+        return SyftError(message=f"Pool does not exist. name={pool_name}")
+
+    try:
+        print(f"Scaling pool name={pool_name} to replicas={replicas}")
+        runner.scale_pool(pool_name=pool_name, replicas=replicas)
+    except Exception as e:
+        return SyftError(message=f"Failed to scale workers {e}")
+
+    return runner.get_pods(pool_name=pool_name)
+
+
+def run_workers_in_kubernetes(
+    worker_image: SyftWorkerImage,
+    worker_count: int,
+    pool_name: str,
+    queue_port: int,
+    start_idx=0,
+    debug: bool = False,
+    username: Optional[str] = None,
+    password: Optional[str] = None,
+    registry_url: Optional[str] = None,
+    **kwargs,
+) -> Union[List[ContainerSpawnStatus], SyftError]:
+    spawn_status = []
+    runner = KubernetesRunner()
+
+    if start_idx == 0:
+        pool_pods = create_kubernetes_pool(
+            runner,
+            worker_image,
+            pool_name,
+            worker_count,
+            queue_port,
+            debug,
+        )
+    else:
+        pool_pods = scale_kubernetes_pool(
+            runner,
+            pool_name,
+            worker_count,
+        )
+        # only interested in the new pods
+        pool_pods = pool_pods[start_idx:]
+
+    # create worker object
+    for pod in pool_pods:
```

Comment:
pool_pods can be SyftError, need to handle that

Reply:
great catch - will fix this

Reply:
pushed a commit